planner/core/resolve_indices.go

```diff
@@ -632,13 +635,60 @@ func (p *baseSchemaProducer) ResolveIndices() (err error) {
 	return
 }
 
+// ResolveIndices implements Plan interface.
+func (p *basePhysicalJoin) ResolveIndices() (err error) {
+	for _, child := range p.children {
+		err = child.ResolveIndices()
+		if err != nil {
+			return err
+		}
+	}
+	switch p.JoinType {
```

Comment:
Why do we need to check outer join branches here? only inner join branch is enough?